Sources/PCReadCursorDebouncerManager.swift

```diff
@@ -0,0 +1,81 @@
+import Foundation
+
+class PCReadCursorDebouncerManager {
+    private var roomIdsToDebouncers: [Int: PCReadCursorDebouncer] = [:]
+    private weak var currentUser: PCCurrentUser?
+
+    init(currentUser: PCCurrentUser) {
+        self.currentUser = currentUser
+    }
+
+    func set(cursorPosition position: Int, inRoomId roomId: Int, completionHandler: @escaping PCErrorCompletionHandler) {
+        if let debouncer = roomIdsToDebouncers[roomId] {
+            debouncer.set(position: position, completionHandler: completionHandler)
+        } else {
+            let debouncer = PCReadCursorDebouncer(roomId: roomId, currentUser: currentUser)
+            roomIdsToDebouncers[roomId] = debouncer
+            debouncer.set(position: position, completionHandler: completionHandler)
+        }
+    }
+}
+
+class PCReadCursorDebouncer {
+    private var roomId: Int
+    private weak var currentUser: PCCurrentUser?
+    private var interval: TimeInterval
+    private var timer: Timer?
+
+    private var sendReadCursorPayload: (position: Int, completionHandler: PCErrorCompletionHandler)? = nil
+
+    init(roomId: Int, currentUser: PCCurrentUser?, intervalMilliseconds: Int = 500) {
+        self.roomId = roomId
+        self.currentUser = currentUser
+        self.interval = Double(intervalMilliseconds  / 1000)
+    }
+
+    deinit {
+        self.timer?.invalidate()
+        self.timer = nil
+    }
+
+    func set(position: Int, completionHandler: @escaping PCErrorCompletionHandler) {
+        sendReadCursorPayload = (position: position, completionHandler: completionHandler)
+
+        guard timer == nil else { return }
+
+        DispatchQueue.main.async { [weak self] in
+            guard let strongSelf = self else {
+                print("self is nil when setting read cursor timer")
+                return
+            }
+
+            strongSelf.timer = Timer.scheduledTimer(
+                timeInterval: strongSelf.interval,
+                target: strongSelf,
+                selector: #selector(strongSelf.sendReadCursor),
+                userInfo: nil,
+                repeats: false
+            )
+        }
+    }
+
+    @objc func sendReadCursor() {
+        guard let currentUser = self.currentUser else {
+            print("currentUser is nil so can't set read cursor")
+            return
+        }
+
+        guard let payload = self.sendReadCursorPayload else {
+            print("sendReadCursorPayload is nil so can't set read cursor")
+            return
+        }
+
+        currentUser.sendReadCursor(
+            position: payload.position,
+            roomId: roomId,
+            completionHandler: payload.completionHandler
```

Comment:
Do we only want the last completion handler to be called? Or do we want every completion handler to be called? I went with the latter in the JS because then the external interface is the same as it was... not a big deal though.